test_util/test_launch.py

```diff
@@ -0,0 +1,156 @@
+import json
+
+import pytest
+import yaml
+from docopt import DocoptExit
+
+import test_util
+from pkgpanda.util import load_json
+from test_util.helpers import Host
+from test_util.launch import get_launcher, main
+
+
+def check_cli(cmd):
+    assert main(cmd.split(' ')) == 0, 'CLI args failed! {}'.format(cmd)
+
+
+def check_success(capsys, tmpdir, config_str):
+    """
+    Runs through the required functions of a laucnher and then
+    runs through the default usage of the script for a
+    given config path and info path, ensuring each step passes
+    if all steps finished successfully, this parses and returns the generated
+    info JSON and stdout description JSON for more specific checks
+    """
+    # Test launcher directly first
+    config = yaml.load(config_str)
+    launcher = get_launcher(config['type'], config['provider_info'])
+    info = launcher.create(config)
+    launcher.wait(info)
+    launcher.describe(info)
+    launcher.delete(info)
+
+    # add config to disk and make info path for CLI testing
+    config_path = tmpdir.join('my_specific_config.yaml')  # test non-default name
+    config_path.write(config_str)
+    config_path = str(config_path)
+    info_path = str(tmpdir.join('my_specific_info.json'))  # test non-default name
+
+    # Now check launcher via CLI
+    check_cli('create --config-path={} --info-path={}'.format(config_path, info_path))
+    # use the info written to disk to ensure JSON parsable
+    info = load_json(info_path)
+    # General assertions about info
+    assert 'type' in info
+    assert 'provider_info' in info
+
+    check_cli('wait --info-path={}'.format(info_path))
+
+    # clear stdout capture
+    capsys.readouterr()
+    check_cli('describe --info-path={}'.format(info_path))
+    # capture stdout from describe and ensure JSON parse-able
+    description = json.loads(capsys.readouterr()[0])
+
```

Comment:
Both of these situations are covered with this test I added: https://github.com/mellenburg/dcos/blob/dcos-launch/test_util/test_launch.py#L249